### PR TITLE
Content addressability dummy PR

### PR DESCRIPTION
**Pull request by aaronlehmann**
I'm creating this PR that tracks the layers-rebased branch just to make it easier to follow discussion on the commits in the branch.

### CODE REVIEW

File: daemon/container.go [hunk 1]
```diff
@@ -116,6 +120,13 @@ func (container *Container) toDisk() error {
 		return err
 	}
 
+	// todo: where was this handled before
```
Comment (tonistiigi) [inline daemon/container.go:123]: That is bit bad todo item. Any ideas?

File: daemon/daemon.go [hunk 2]
```diff
@@ -713,8 +726,25 @@ func NewDaemon(config *Config, registryService *registry.Service) (daemon *Daemo
 		return nil, err
 	}
 
-	logrus.Debug("Creating images graph")
-	g, err := graph.NewGraph(filepath.Join(config.Root, "graph"), d.driver, uidMaps, gidMaps)
+	imageRoot := filepath.Join(config.Root, "image", d.driver.String())
+	if err := os.MkdirAll(filepath.Join(imageRoot, "layerdb"), 0600); err != nil {
+		return nil, err
+	}
+	fms := layer.NewFileMetadataStore(filepath.Join(imageRoot, "layerdb"))
```
Comment (tonistiigi) [inline daemon/daemon.go:733]: `NewFileMetadataStore` requires root to be created by the client, does not init itself.
`NewFSStoreBackend` does init in constructor, may error.
`NewFSMetadataStore` doesn't init, may fail on first insertion.

We should make them all use a similar pattern. Also maybe combine the names, at least trivial stuff like `File` vs `FS`.

File: daemon/daemon.go [hunk 3]
```diff
@@ -736,23 +766,29 @@ func NewDaemon(config *Config, registryService *registry.Service) (daemon *Daemo
...
-	logrus.Debug("Creating repository list")
-	tagCfg := &graph.TagStoreConfig{
-		Graph:    g,
-		Key:      trustKey,
-		Registry: registryService,
-		Events:   eventsService,
-	}
-	repositories, err := graph.NewTagStore(filepath.Join(config.Root, "repositories-"+d.driver.String()), tagCfg)
+
+	tagStore, err := tag.NewTagStore(filepath.Join(imageRoot, "repositories.json"))
```
Comment (tonistiigi) [inline daemon/daemon.go:773]: Do we want to keep calling this `repositories.json`? ... do we want to call the manager tagStore if it actually doesn't have much to do with tags any more.

Comment (aaronlehmann) [inline daemon/daemon.go:773]: Maybe tags.json?

**Pull request closed without merging**